src/Ui/Table.elm

```diff
@@ -0,0 +1,464 @@
+module Ui.Table exposing
+    ( ColumnAlignment(..)
+    , ColumnConfig
+    , Flex(..)
+    , Model
+    , Msg
+    , SortDirection(..)
+    , TableConfig
+    , addColumn
+    , addExtendableView
+    , cellColumns
+    , cellCustom
+    , cellFloat
+    , cellInt
+    , cellLink
+    , cellRows
+    , cellText
+    , column
+    , columnWithDataEntryId
+    , defaultConfig
+    , hideColumn
+    , setColumnAlignment
+    , setColumnWidth
+    , showColumn
+    , update
+    , view
+    )
+
+import Css
+import Css.Transitions exposing (transition)
+import Dict exposing (Dict)
+import Html.Styled exposing (Attribute, Html, a, div, span)
+import Html.Styled.Attributes exposing (css, href)
+import Html.Styled.Events exposing (onClick)
+import Html.Styled.Keyed exposing (lazyNode3)
+import List
+import Maybe.Extra exposing (isJust)
+import Ui.Css.Palette as Palette
+import Ui.Css.TextStyle exposing (toCssStyle)
+import Ui.Icon
+import Ui.Palette
+import Ui.Text as Text
+import Ui.TextStyle as TextStyle exposing (TextStyle)
+
+
+type SortDirection
+    = Ascending
+    | Descending
+
+
+type ColumnAlignment
```

Comment:
isn't it better to do this in the ColumnContent html?

Reply:
Probably, I'll try it out!